src/Guide/Api/Methods.hs

```diff
@@ -65,53 +62,46 @@ createCategory db title' group' = do
 
 -- | Edit categoty's note.
 setCategoryNotes :: DB -> Uid Category -> CTextEdit -> Handler NoContent
-setCategoryNotes db catId CTextEdit{..} =
-  dbQuery db (GetCategoryMaybe catId) >>= \case
-    Nothing -> throwError (err404 {errBody = "Category not found"})
-    Just Category{..} -> do
-      let serverModified = markdownBlockMdSource _categoryNotes
-      let original = unH cteOriginal
-      let modified = unH cteModified
-      if original /= serverModified then do
-        let merged = merge original modified serverModified
-        let conflict = CMergeConflict
-                { cmcOriginal = cteOriginal
-                , cmcModified = cteModified
-                , cmcServerModified = H serverModified
-                , cmcMerged = H merged
-                }
-        throwError (err409 {errBody = encode conflict})
-      else do
-        (_edit, _newCategory) <- dbUpdate db (SetCategoryNotes catId modified)
-        pure NoContent
+setCategoryNotes db catId CTextEdit{..} = do
+  serverModified <- markdownBlockMdSource . _categoryNotes <$> getCategoryOrFail db catId
+  let original = unH cteOriginal
```

Comment:
When I said that this piece of code can be abstracted away, I meant the whole "if original /= serverModified then ... throwError ...".